extensions/resource-deployment/src/services/platformService.ts

```diff
@@ -2,38 +2,65 @@
  *  Copyright (c) Microsoft Corporation. All rights reserved.
  *  Licensed under the Source EULA. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
-
+import * as azdata from 'azdata';
 import * as fs from 'fs';
+import * as cp from 'promisify-child-process';
+import * as sudo from 'sudo-prompt';
 import * as vscode from 'vscode';
-import * as azdata from 'azdata';
-import * as cp from 'child_process';
+import * as nls from 'vscode-nls';
+import { OsType } from '../interfaces';
 import { getErrorMessage } from '../utils';
 
+const localize = nls.loadMessageBundle();
+const extensionOutputChannel = localize('resourceDeployment.outputChannel', 'Deployments');
+const sudoPromptTitle = 'AzureDataStudio';
 /**
  * Abstract of platform dependencies
  */
 export interface IPlatformService {
+	osType(): OsType;
 	platform(): string;
 	storagePath(): string;
 	copyFile(source: string, target: string): Promise<void>;
 	fileExists(file: string): Promise<boolean>;
 	openFile(filePath: string): void;
-	showErrorMessage(message: string): void;
+	showErrorMessage(error: Error | string): void;
+	logToOutputChannel(data: string | Buffer, header?: string): void;
+	outputChannelName(): string;
+	showOutputChannel(preserveFocus?: boolean): void;
 	isNotebookNameUsed(title: string): boolean;
 	makeDirectory(path: string): Promise<void>;
+	ensureDirectoryExists(directory: string): Promise<void>;
 	readTextFile(filePath: string): Promise<string>;
 	runCommand(command: string, options?: CommandOptions): Promise<string>;
 	saveTextFile(content: string, path: string): Promise<void>;
 	deleteFile(path: string, ignoreError?: boolean): Promise<void>;
 }
 
+interface CommandOutput {
+	stdout: string;
+	stderr: string;
+}
+
 export interface CommandOptions {
 	workingDirectory?: string;
 	additionalEnvironmentVariables?: NodeJS.ProcessEnv;
+	sudo?: boolean;
+	commandTitle?: string;
+	ignoreError?: boolean;
 }
 
+/**
+ * A class that provides various services to interact with the platform on which the code runs
+ */
 export class PlatformService implements IPlatformService {
+
+	private _outputChannel: vscode.OutputChannel = vscode.window.createOutputChannel(extensionOutputChannel);
+
 	constructor(private _storagePath: string = '') {
+		if (_storagePath) {
+			this.ensureDirectoryExists(_storagePath);
```

Comment:
I'd not do this here, not good for testing, during unit testing, we don't might use this but we don't want to actually do the folder creation.

Reply:
for unit testing, can't we pass in not storagePath?